build: Windows cross Qt broken on macOS host

```bash
make -C depends HOST=x86_64-w64-mingw32
<snip>
-- Configure summary:

Building for: win32-g++ (x86_64, CPU features: )
Compiler: gcc 14.2.0
Build options:
  Mode ................................... release
  Optimize release build for size ........ no
  Fully optimize release builds (-O3) .... no
  Building shared libraries .............. no
  Unity Build ............................ no
  Relocatable ............................ no
  Using precompiled headers .............. yes
  Using Link Time Optimization (LTCG) .... no
  Using Intel CET ........................ no
  Target compiler supports:
    x86 Intrinsics ....................... Basic VAES AVX512VBMI2
  Sanitizers:
    Addresses ............................ no
    Threads .............................. no
    Memory ............................... no
    Fuzzer (instrumentation only) ........ no
    Undefined ............................ no
  Build parts ............................ libs
  Install examples sources ............... no
  App store compliance ................... no
Qt modules and options:
  Qt Concurrent .......................... no
  Qt D-Bus ............................... no
  Qt D-Bus directly linked to libdbus .... no
  Qt Gui ................................. yes
  Qt Network ............................. yes
  Qt PrintSupport ........................ no
  Qt Sql ................................. no
  Qt Testlib ............................. yes
  Qt Widgets ............................. yes
  Qt Xml ................................. yes
Support enabled for:
  Using pkg-config ....................... no
  Using vcpkg ............................ no
  udev ................................... no
  OpenSSL ................................ no
    Qt directly linked to OpenSSL ........ no
  OpenSSL 1.1 ............................ no
  OpenSSL 3.0 ............................ no
  Using system zlib ...................... no
  Zstandard support ...................... no
  Thread support ......................... yes
Common build options:
  Linker can resolve circular dependencies  yes
Qt Core:
  backtrace .............................. no
  DoubleConversion ....................... yes
    Using system DoubleConversion ........ no
  GLib ................................... no
  ICU .................................... no
  Using system libb2 ..................... yes
  Built-in copy of the MIME database ..... yes
  Application permissions ................ yes
  Tracing backend ........................ <none>
  OpenSSL based cryptographic hash ....... no
  Logging backends:
    journald ............................. no
    syslog ............................... no
    slog2 ................................ no
  PCRE2 .................................. yes
    Using system PCRE2 ................... no
Qt Network:
  getifaddrs() ........................... no
  IPv6 ifname ............................ no
  libproxy ............................... no
  Schannel ............................... no
  DTLS ................................... no
  OCSP-stapling .......................... no
  SCTP ................................... no
  Use system proxies ..................... no
  GSSAPI ................................. no
  Brotli Decompression Support ........... no
  qIsEffectiveTLD() ...................... no
    Built-in publicsuffix database ....... no
    System publicsuffix database ......... no
Core tools:
  Android deployment tool ................ no
  macOS deployment tool .................. no
  Windows deployment tool ................ no
  qmake .................................. no
Qt Gui:
  Accessibility .......................... yes
  FreeType ............................... no
    Using system FreeType ................ no
  HarfBuzz ............................... yes
    Using system HarfBuzz ................ no
  Fontconfig ............................. no
  Image formats:
    GIF .................................. no
    ICO .................................. no
    JPEG ................................. no
      Using system libjpeg ............... no
    PNG .................................. yes
      Using system libpng ................ no
  Text formats:
    HtmlParser ........................... yes
    CssParser ............................ yes
    OdfWriter ............................ no
    MarkdownReader ....................... no
      Using system libmd4c ............... no
    MarkdownWriter ....................... no
  EGL .................................... no
  OpenVG ................................. no
  OpenGL:
    Desktop OpenGL ....................... no
    Dynamic OpenGL ....................... no
    OpenGL ES 2.0 ........................ no
    OpenGL ES 3.0 ........................ no
    OpenGL ES 3.1 ........................ no
    OpenGL ES 3.2 ........................ no
  Vulkan ................................. no
  Session Management ..................... no
Features used by QPA backends:
  evdev .................................. no
  libinput ............................... no
  HiRes wheel support in libinput ........ no
  INTEGRITY HID .......................... no
  mtdev .................................. no
  tslib .................................. no
  xkbcommon .............................. no
  X11 specific:
    XLib ................................. no
    XCB Xlib ............................. no
    EGL on X11 ........................... no
    xkbcommon-x11 ........................ no
    xcb-sm ............................... no
QPA backends:
  DirectFB ............................... no
  EGLFS .................................. no
  EGLFS details:
    EGLFS OpenWFD ........................ no
    EGLFS i.Mx6 .......................... no
    EGLFS i.Mx6 Wayland .................. no
    EGLFS RCAR ........................... no
    EGLFS EGLDevice ...................... no
    EGLFS GBM ............................ no
    EGLFS VSP2 ........................... no
    EGLFS Mali ........................... no
    EGLFS Raspberry Pi ................... no
    EGLFS X11 ............................ no
  LinuxFB ................................ no
  VNC .................................... no
  VK_KHR_display ......................... no
  QNX:
    lgmon ................................ no
    IMF .................................. no
  XCB:
    Using system-provided xcb-xinput ..... no
    GL integrations:
      GLX Plugin ......................... no
        XCB GLX .......................... no
      EGL-X11 Plugin ..................... no
  Windows:
    Direct 2D ............................ yes
    Direct 2D 1.1 ........................ yes
    DirectWrite .......................... yes
    DirectWrite 3 ........................ yes
Qt Widgets:
  GTK+ ................................... no
  Styles ................................. Fusion Windows WindowsVista
Qt Testlib:
  Tester for item models ................. yes
  Batch tests ............................ no
 

Note: Using static linking will disable the use of dynamically loaded plugins. Make sure to import all needed static plugins, or compile needed modules into the library.

WARNING: Basic cpp/winrt support missing. Some features might not be available.

-- 

Qt is now configured for building. Just run 'cmake --build . --parallel'

Once everything is built, you must run 'cmake --install .'
Qt will be installed into '/bitcoin/depends/x86_64-w64-mingw32'

To configure and build other Qt modules, you can use the following convenience script:
        /bitcoin/depends/x86_64-w64-mingw32/bin/qt-configure-module

If reconfiguration fails for some reason, try removing 'CMakeCache.txt' from the build directory
Alternatively, you can add the --fresh flag to your CMake flags.

-- Configuring done (30.4s)
CMake Error in qtbase/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "zstd::libzstd_shared"
  configuration "Release".


CMake Error in qtbase/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "zstd::libzstd_shared"
  configuration "Release".


CMake Error in qtbase/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "zstd::libzstd_shared"
  configuration "Release".


CMake Error in qtbase/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "zstd::libzstd_shared"
  configuration "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target
  "double-conversion::double-conversion" configuration "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target
  "double-conversion::double-conversion" configuration "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target
  "double-conversion::double-conversion" configuration "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target
  "double-conversion::double-conversion" configuration "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "md4c::md4c" configuration
  "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "md4c::md4c" configuration
  "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "md4c::md4c" configuration
  "Release".


CMake Error in qtbase/src/CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "md4c::md4c" configuration
  "Release".


-- Generating done (0.7s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
CMake Error at qtbase/cmake/QtProcessConfigureArgs.cmake:1077 (message):
  CMake exited with code 1.
```

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -190,6 +190,14 @@ ifneq ($(host),$(build))
 $(package)_cmake_opts += -DCMAKE_SYSTEM_NAME=$($(host_os)_cmake_system_name)
 $(package)_cmake_opts += -DCMAKE_SYSTEM_VERSION=$($(host_os)_cmake_system_version)
 $(package)_cmake_opts += -DCMAKE_SYSTEM_PROCESSOR=$(host_arch)
+# Native packages cannot be used during cross-compiling. However,
+# Qt still unconditionally tries to find them, which causes issues
+# in some cases, such as when cross-compiling from macOS to Windows.
+# Explicitly disable this unnecessary Qt behaviour.
+$(package)_cmake_opts += -DCMAKE_DISABLE_FIND_PACKAGE_Libb2=TRUE
+$(package)_cmake_opts += -DCMAKE_DISABLE_FIND_PACKAGE_WrapSystemDoubleConversion=TRUE
+$(package)_cmake_opts += -DCMAKE_DISABLE_FIND_PACKAGE_WrapSystemMd4c=TRUE
+$(package)_cmake_opts += -DCMAKE_DISABLE_FIND_PACKAGE_WrapZSTD=TRUE
 endif
 ifeq ($(host_os),darwin)
 $(package)_cmake_opts += -DCMAKE_INSTALL_NAME_TOOL=true

